sdk/test-utils/recorder/src/matcher.ts

```diff
@@ -5,13 +5,41 @@ import { createPipelineRequest, HttpClient } from "@azure/core-rest-pipeline";
 import { paths } from "./utils/paths";
 import { RecorderError } from "./utils/utils";
 
-export type Matcher = "HeaderlessMatcher" | "BodilessMatcher";
+export type Matcher = "HeaderlessMatcher" | "BodilessMatcher" | "CustomDefaultMatcher";
+
+/**
+ * Body the customer matcher expects.
+ */
+export interface CustomMatcherBody {
```

Comment:
What do you think about `CustomMatcherOptions`?

Reply:
OK